File: leetcode/go/191.number-of-1-bits.go
/*
 * @lc app=leetcode id=191 lang=golang
 *
 * [191] Number of 1 Bits
 */

// @lc code=start
func hammingWeight(num uint32) int {
	ret := 0
	for num > 0 {
		num = num & (num - 1)
		ret++
	}
	return ret
}

// @lc code=end

File: leetcode/cpp/191.number-of-1-bits.cpp
/*
 * @lc app=leetcode id=191 lang=cpp
 *
 * [191] Number of 1 Bits
 */
#include<iostream>
#include <vector>
using namespace std;
// @lc code=start
class Solution {
public:
    int hammingWeight(int n) {
        int ret = 0;
        while ( n > 0 ){
            n = n& (n - 1 );
            ret = ret + 1;
        }
        return ret;
        
    }
};
// @lc code=end

